src/func.d

```diff
@@ -2169,13 +2169,12 @@ public:
             sc2.callSuper = 0;
             sc2.pop();
         }
-        if (needsClosure())
+
+        if (checkClosure())
         {
-            if (setGC())
-                error("@nogc function allocates a closure with the GC");
-            else
-                printGCUsage(loc, "using closure causes GC allocation");
+            //errors = true;
```

Comment:
Could you put a short version of that explanation in the comment instead?
eg `// We should be setting errors here instead of relying on the global error count`
The meaning of dead code as comments unfortunately tends to get lost over time.